Sources/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScope.swift

```diff
@@ -93,16 +97,21 @@ internal class RUMUserActionScope: RUMScope, RUMContextProvider {
     // MARK: - RUMScope
 
     func process(command: RUMCommand, context: DatadogV1Context, writer: Writer) -> Bool {
-        if let expirationTime = possibleExpirationTime(currentTime: command.time), allResourcesCompletedLoading() {
-            sendActionEvent(completionTime: expirationTime, on: nil, context: context, writer: writer)
-            return false
-        }
+        // TODO - Handle time expiration gracefully now that delays have been introduced
+//        if let expirationTime = possibleExpirationTime(currentTime: command.time), allResourcesCompletedLoading() {
+//            sendActionEvent(completionTime: expirationTime, on: nil, context: context, writer: writer)
+//            return false
+//        }
 
         lastActivityTime = command.time
         switch command {
         case is RUMStopViewCommand:
             sendActionEvent(completionTime: command.time, on: command, context: context, writer: writer)
             return false
+        case let command as RUMAddUserActionCommand:
+            isRage = command.isRage
+            sendActionEvent(completionTime: command.time, on: command, context: context, writer: writer)
+            return false
```

Comment:
The `RUMAddUserActionCommand` shouldn't be processed by an already opened action-scope. Instead, the `isRage` can be set when initialising the discrete action scope from the `RUMViewScope`.